modules/claims_api/app/controllers/claims_api/v0/forms/intent_to_file_controller.rb

```diff
@@ -9,20 +9,28 @@ class IntentToFileController < ClaimsApi::BaseFormController
         skip_before_action(:authenticate)
         before_action :validate_json_format, if: -> { request.post? }
         before_action :validate_json_schema, only: %i[submit_form_0966 validate]
-        before_action :check_future_type, only: [:submit_form_0966]
 
         FORM_NUMBER = '0966'
 
         def submit_form_0966
-          response = itf_service.create_intent_to_file(form_type)
-          render json: response['intent_to_file'],
+          bgs_response = bgs_service.intent_to_file.insert_intent_to_file(intent_to_file_options)
+          render json: bgs_response,
                  serializer: ClaimsApi::IntentToFileSerializer
         end
 
         def active
-          response = itf_service.get_active(active_param)
-          render json: response['intent_to_file'],
-                 serializer: ClaimsApi::IntentToFileSerializer
+          bgs_response = bgs_service.intent_to_file.find_intent_to_file_by_ptcpnt_id_itf_type_cd(
+            target_veteran.participant_id,
+            ClaimsApi::IntentToFile::ITF_TYPES[active_param]
+          )
+          if bgs_response.present?
+            bgs_active = bgs_response.detect do |itf|
+              itf.present? && itf[:itf_status_type_cd] == 'Active' && itf[:exprtn_dt].to_datetime > Time.zone.now
+            end
+            render json: bgs_active, serializer: ClaimsApi::IntentToFileSerializer
```

Comment:
Doesnt this risk passing nil to the serializer? You could get a valid BGS response for a veteran who is found and not have an ITF for that compensation type. Or it doesnt meet the date parameters. Then nil is passed to the serializer and the result would be a 500. In that case I think a 404 would be a more accurate response.

Reply:
It'll filter by type yes but good catch on the bgs_active state, refactored it so that it shouldn't be an issue any further. Thanks.